coinbaseadvanced/client.py

```diff
@@ -642,31 +643,38 @@ def get_product_candles_all(
         Gets all requested product candles
         """
 
-        # pre-calculate granularity entries in minutes 
-        granularity_minutes = timedelta(minutes=GRANULARITY_MAP_IN_MINUTES[granularity.value])
-        # request size of 300 (max allowed by coinbase)
-        granularity_minutes_x300 = granularity_minutes * 300
+        # step_size: pre-calculate granularity entries in minutes.
+        step_size_in_mins = timedelta(minutes=GRANULARITY_MAP_IN_MINUTES[granularity.value])
+
+        # Max amount of candles that can be returned.
+        # Coinbase API enforcement/error if you try to retrieve >= 300 below:
+        # "start and end argument is invalid - number of candles requested should be less than 300."
+        max_candles_amount = 299
+
+        # request size of 299 (max allowed by coinbase)
+        time_window_in_mins = step_size_in_mins * max_candles_amount
```

Comment:
The previous code had a bug since it tried to retrieve 300 instead of 299. Fixed here. Also simplified the logic a bit.

Reply:
Do you have a reference to where the 299 limit is documented? I got 300 from the error message returned when requesting more than 300.